scripts/nb-tester/test-notebook.py

```diff
@@ -39,6 +41,39 @@ class ExecuteOptions:
     submit_jobs: bool
 
 
+@dataclass(frozen=True)
+class NotebookWarning:
+    cell_index: int
+    msg: str
+
+    def report(self):
+        message = f"Warning detected in cell {self.cell_index}:\n"
+        for line in self.msg.splitlines():
+            message += (
+                textwrap.fill(
+                    line, width=77, initial_indent=" │ ", subsequent_indent=" │ "
+                )
+                + "\n"
+            )
+        print(f"\033[0;33m{message}\033[0m", flush=True)
```

Comment:
It'd be good to de-magic the escape codes by creating class CONSTANTS that say what they correspond to. Another approach is to create a helper function like this:

https://github.com/pantsbuild/pants/blob/e7b4b251b74f5e8332e6d527d0c7a5a3ffc878ab/build-support/migration-support/convert_source_to_sources.py#L95-L96